R/SingleR_Seurat.R

```diff
@@ -221,6 +222,35 @@ RunSingleR <- function(seuratObj = NULL, datasets = c('hpca', 'blueprint', 'dice
     }
 
     Tabulate_SingleR(seuratObj, plotIndividually = TRUE, datasets = datasets)
+
+    # Create a pseudo consensus:
+    if (createConsensus) {
+      allFields <- names(seuratObj@meta.data)
+      allFields <- allFields[grepl(allFields, pattern = 'label')]
+
+      fieldsToUse <- allFields[!grepl(allFields, pattern = 'fine')]
+      dat <- seuratObj@meta.data[fieldsToUse]
+
+      for (colName in names(dat)) {
+        dat[[colName]][grepl(dat[[colName]], pattern = "Myeloid|Monocyte|Macrophage", ignore.case = T)] <- "Myeloid"
+        dat[[colName]][grepl(dat[[colName]], pattern = "Tcell|T_cell|T cell|T-cell|TCell|T_Cell|T Cell|T-Cell|CD8|CD4|Treg", ignore.case = T)] <- "NK/T_cell"
+        dat[[colName]][grepl(dat[[colName]], pattern = "NKcell|NK_cell|NK cell|NK-cell|NKCell|NK_Cell|NK Cell|NK-Cell", ignore.case = T)] <- "NK/T_cell"
+        dat[[colName]][grepl(dat[[colName]], pattern = "Bcell|B_cell|B cell|B-cell|BCell|B_Cell|B Cell|B-Cell", ignore.case = T)] <- "B_cell"
+        dat[[colName]][grepl(dat[[colName]], pattern = "DC|Dendritic")] <- "DCs"
```

Comment:
I think this is fine for our scope, but for the top label hpca/dice/monaco/blueprint all have no conflict with grepl-ing "dc". 

We could include ignore.case = T here if desired or we could commit to sanitizing any other callers ("dc" -> "DC") if we recycle this code and add more callers to the consensus.